tests/rptest/clients/kafka_cli_tools.py

```diff
@@ -247,6 +250,172 @@ def get_api_versions(self):
+        if operation == "execute":
+            args.append("--execute")
+        elif operation == "verify":
+            args.append("--verify")
+        else:
+            raise NotImplementedError(f"Unknown operation: {operation}")
+
+        return self._run("kafka-reassign-partitions.sh", args)
+
+    def get_topics_and_partitions(self, reassignments: dict):
+        topic_names = []
+        partition_idxs = []
+        for reassignment in reassignments["partitions"]:
+            topic_names.append(reassignment["topic"])
+            partition_idxs.append(reassignment["partition"])
+
+        return topic_names, partition_idxs
+
+    def execute_reassign_partitions(self, reassignments: dict):
+        output = self.reassign_partitions(reassignments=reassignments,
+                                          operation="execute")
+
+        topic_names, partition_idxs = self.get_topics_and_partitions(
+            reassignments)
+
+        # Check output
+        # Output has the following structure
+        # Line 0 - "Current partition replica assignment"
+        # Line 1 - empty
+        # Line 2 - Json structure showing the current replica sets for each partition
+        # Line 3 - empty
+        # Line 4 - "Save this to use as the --reassignment-json-file option during rollback"
+        # Line 5 - "Successfully started partition reassignments for topic1-p0,topic1-p1,...,topic2-p0,topic2-p1,..."
+        lines = output.splitlines()
+
+        # Check the output in reverse order
+
+        # The last line should list topic partitions
+        line = lines.pop().strip()
+        assert line.startswith(
+            "Successfully started partition reassignments for ")
+        topic_partitions = line.removeprefix(
+            "Successfully started partition reassignments for ").split(',')
+        tp_re = re.compile(r"^(?P<topic>[a-z\-]+?)-(?P<pid>[0-9]+?)$")
+        for tp in topic_partitions:
+            tp_match = tp_re.match(tp)
+            self._redpanda.logger.debug(
+                f"topic partition match: {tp}, {tp_match}")
+            assert tp_match is not None
+            assert tp_match.group("topic") in topic_names
+            assert int(tp_match.group("pid")) in partition_idxs
+
+        # The next lines are exact strings
+        assert lines.pop().strip(
+        ) == "Save this to use as the --reassignment-json-file option during rollback"
+        assert len(lines.pop()) == 0
+
+        # Then a json structure
+        current_assignment = json.loads(lines.pop().strip())
+        assert type(current_assignment) == type({}), "Expected JSON object"
+
+        # Then another set of exact strings
+        assert len(lines.pop()) == 0
+        assert lines.pop().strip() == "Current partition replica assignment"
+
+        if len(lines) != 0:
+            raise RuntimeError(f"Unexpected output: {lines}")
+
+        return output
+
+    def verify_reassign_partitions(self, reassignments: dict):
```

Comment:
To me, feels like that most methods in the `KafkaCLITools` class are responsible for making a request to an underlying tool, then returning that output nicely parsed. I think these two methods do a bit more then that comparing expected to observed output , some custom logic that IMO would probably belong in the reassign partitions test class.